sdk/keyvault/azure-keyvault-secrets/azure/keyvault/secrets/_shared/challenge_auth_policy.py

```diff
@@ -74,6 +72,11 @@ def _get_challenge_request(request):
 class ChallengeAuthPolicy(ChallengeAuthPolicyBase, HTTPPolicy):
     """policy for handling HTTP authentication challenges"""
 
+    def __init__(self, credential, **kwargs):
+        # type: (TokenCredential, **Any) -> None
+        self._credential = credential
```

Comment:
I guess this means every release of KV now will need this new azure-identity, be sure to update the setup.py to reflect that

Reply:
Assuming you mean KV will need the new azure-core because the async challenge auth policy uses `AsyncTokenCredential`, this is what @xiangyan99 and I have been discussing. I haven't bumped dependencies because `AsyncTokenCredential` is only used in type annotations. I think a user combining old azure-core with new azure-keyvault would need to run mypy or pylint on azure-keyvault to see a warning. Is that mistaken, or too sloppy?